.github/workflows/preview-docs.yml

```
@@ -0,0 +1,96 @@
+name: Preview Documentation
+on:
+  pull_request:
+  pull_request_target:
+    types: [labeled]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  doc:
+    name: Build Documentation
+    runs-on: ubuntu-20.04
```

Our idea is to allow this CI to run when labeled like `PreviewDocumentation-accepted`.
```suggestion
    runs-on: ubuntu-20.04
    if: contains(github.event.pull_request.labels.*.name, 'PreviewDocumentation-accepted')
```